docker-compose-net/docker-compose-net.md

```diff
@@ -0,0 +1,234 @@
+ entirely up to you):
+ 
+ **1.** The Weave Net Docker API Proxy. See [Integrating Docker via the API Proxy](https://www.weave.works/docs/net/latest/weave-docker-api/).  
+ 
+ **2.**  The Docker Network Plugin framework. The Docker Network Plugin is used when 
+ Docker containers are started with the --net flag, for example: 
+ 
+ `docker run --net <docker-run-options>`
+ 
+ **Where,** 
+ 
+  * `<docker-run-options>` are the [docker run options](https://docs.docker.com/engine/reference/run/) 
+  you give to your container on start 
+ 
+ >>**Note:** If a Docker container is started with the `--net` flag, then the Weave Docker API Proxy
+ is automatically disabled and is not used to attach containers. See [Integrating Docker via the Network Plugin](https://www.weave.works/docs/net/latest/features/#plugin).
+
+
+##<a name="setup-proxy"></a>Docker Compose and the Weave Net Docker API Proxy
+
+
+The following is a simple setup that shows how to configure docker-compose for Weave Net across two hosts using the Docker API proxy. You will launch the alpine using Docker Compose and then run 'ping' to test the connectivity between the two hosts. 
+
+**1.** Create two hosts on Virtualbox with docker machine in two different terminal windows: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-01
+~~~
+
+And, in a new terminal window, create the other VM: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-02
+~~~
+
+**2.** In order to peer hosts on Weave Net, the IP address of one of the VMs you just created must be known beforehand: 
+
+~~~bash
+docker-machine ip weave-compose-02
+~~~
+
+Make a note of the IP address that appears. 
+
+**3.** Install and then launch Weave Net onto both VMs. From one terminal windown launch Weave: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch
+~~~
+
+And in a second terminal launch Weave onto the other host, and pass the IP of the first host to it: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch <ip address host-02>
+~~~
+
+**4.** Setup the environment to run the Weave Docker API proxy: 
+
+~~~bash
+eval $(weave env)
+~~~
+
+With the Weave environment set up, you can run regular Docker commands on the same command line from which you created VMs. If you are attaching containers to the Weave Network, this step is mandatory, unless you've opted to use the Weave Net Docker Plugin which is described below. 
+
+**5.** Run the Docker Compose file and direct each service to a host: 
+
+~~~bash
+cd proxy-compose
+docker-compose up -d pingme
+~~~
+
+Next, switch to the second terminal, and change to the compose-plugin sub-directory and run the compose file:
+
+~~~bash
+cd proxy-compose
+docker-compose up -d pinger 
+~~~
+
+6. If all has went well, you should see 'Hello, Weave!' printed on your command line. Pingme will keep resonding to the command sent by Pinger for 3600 seconds. 
```

Comment:
The user won't see anything at all because they ran `docker-compose up -d` which puts the command in the background.  You need to tell them to run `docker ps` and `docker logs <container-id-of-pinger>` or similar. Or maybe you can predict the container name that compose will create and do something like `docker logs proxy_compose_pinger_1` or something (check this).